### PR TITLE
DCRM-133: Update backend exercise

### PR DESCRIPTION
**Pull request by Y3K**

### CODE REVIEW

File: backend/rest-api/README.md [hunk 1]
```diff
@@ -1,55 +1,128 @@
...
+This project is meant to examine your ability to call 3rd party data services
+and handle the response from these services.
 
 ## Project
 
-Consider the following JSON REST API with a single list endpoint:
+Your task is to build and deliver an API with the following requirements:
+
+- [ ] A new git repo with your project's code base.
+- [ ] A runnable Python project with the following specifications:
```
Comment (aaronmarkey) [inline backend/rest-api/README.md]: Add a list item for a `README` specifying how to run the project, how to test, etc.

File: backend/rest-api/README.md [hunk 2]
```diff
@@ -1,55 +1,128 @@
...
+  - [ ] Listen to HTTP requests.
+  - [ ] Has a single endpoint `/aggregated-data/`.
+  - [ ] Requests https://chaos.leaflink.com/ for data.
```
Comment (aaronmarkey) [inline backend/rest-api/README.md]: Did we settle on this domain name?

Comment (Y3K) [reply backend/rest-api/README.md]: Well, it was suggested and I think it fits.

File: backend/rest-api/README.md (hunk 1 above)
Comment (Y3K) [reply backend/rest-api/README.md]: That is specified in the `deliverables` section.

File: backend/rest-api/README.md [hunk 3]
```diff
@@ -1,55 +1,128 @@
...
+### Average Languages per Country by subregion
+
+`GET /aggregated-data/?aggregation=avg&field=languages&by=subregion`
+
+```json
+{
+  "Australia and New Zealand": 1.2,
+  "Caribbean": 1.29,
+  "Central America": 1.12,
+  "Central Asia": 2.0
```
Comment (aaronmarkey) [inline backend/rest-api/README.md]: Missing a comma here.

File: backend/rest-api/README.md (hunk 1 above)
Comment (aaronmarkey) [reply backend/rest-api/README.md]: Let's clean up the **Projects** and **Deliverables** section. Only include Python requirements. Move the git repo stuff down to **Projects**.

File: backend/rest-api/README.md [hunk 4]
```diff
@@ -1,61 +1,135 @@
...
+
+- [ ] Average
+  - [ ] Area
+  - [ ] Borders per Country
+  - [ ] Currencies per Country
+  - [ ] Gini
+  - [ ] Languages per Country
+  - [ ] Latitude, Longitude
+  - [ ] Population
+  - [ ] Timezone
```
Comment (fission6) [inline backend/rest-api/README.md]: does this make sense to average?

File: backend/rest-api/README.md [hunk 5]
```diff
@@ -1,61 +1,135 @@
...
+### Agreggations
+
+Based on the request GET parameters the endpoint should handle the following
+aggregations, all of them by Region or Subregion:
+
+- [ ] Average
+  - [ ] Area
+  - [ ] Borders per Country
+  - [ ] Currencies per Country
+  - [ ] Gini
```
Comment (fission6) [inline backend/rest-api/README.md]: what the !

File: backend/rest-api/README.md [hunk 6]
```diff
@@ -1,61 +1,135 @@
...
+and handle the response from these services.
 
 ## Project
 
-Consider the following JSON REST API with a single list endpoint:
+Your task is to build and deliver a Python API with the following requirements:
+
+- [ ] Listen to HTTP requests.
+- [ ] Has a single endpoint `/aggregated-data/`.
+- [ ] Requests https://dev-interview-countries.leaflink.com/ for data.
```
Comment (fission6) [inline backend/rest-api/README.md]: I don't think we need to be clever, we can specifically call out this endpoint has been designed to be semi stable and that you should expect to have to handle edge cases like timeouts and !200s

**Comment by fission6** [comment]
Just a heads up I would hand out bonus points to any candidate who only calls the API endpoint once and caches results (for a "sane" period) and takes a position that none of this data changes "frequently"

File: backend/rest-api/README.md (hunk 6 above)
Comment (aaronmarkey) [reply backend/rest-api/README.md]: Agree, lets make it perfectly clear somewhere in this doc that the service has be specifically designed to fail in some cases.

File: backend/rest-api/README.md (hunk 4 above)
Comment (aaronmarkey) [reply backend/rest-api/README.md]: @fission6 Is correct. 

Let's nix all of the Timezone stuff from the agg. data. Here, and min/max.

**Pull request merged**